### PR TITLE
Station Animation Update

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Two simple animation updates

### CODE REVIEW

File: src/OpenLoco/src/Map/StationElement.h [hunk 1]
```diff
@@ -57,12 +58,21 @@ namespace OpenLoco::World
...
+        uint8_t buildingType() const
+        {
+            return (_stationId & 0xFC00) >> 10;
+        }
+        // ((_6 & 0xFC00) >> 10) Note: Only none 0 for airports
+        void setBuildingType(uint8_t type)
         {
             _stationId &= ~0xFC00;
-            _stationId |= (val & 0x3F) << 2;
+            _stationId |= (type & 0x3F) << 10;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Map/StationElement.h:70]: woops. Lets hope this isn't the source of any issues...

File: src/OpenLoco/src/Map/AnimationManager.cpp [hunk 2]
```diff
@@ -62,7 +63,7 @@ namespace OpenLoco::World::AnimationManager
             case 1:
                 return call(0x00479413, regs) & X86_FLAG_CARRY;
             case 2:
-                return call(0x004BD528, regs) & X86_FLAG_CARRY;
+                return false;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Map/AnimationManager.cpp:66]: Seriously? 🥹

File: src/OpenLoco/src/Map/StationElement.h [hunk 3]
```diff
@@ -57,12 +58,21 @@ namespace OpenLoco::World
             _stationId &= ~0x3FF;
             _stationId |= enumValue(id) & 0x3FF;
         }
-        void setUnk7SLR2(uint8_t val)
+        // ((_6 & 0xFC00) >> 10) Note: Only none 0 for airports
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Map/StationElement.h:61]: By `none 0`, do you mean `non-zero`? Just making sure.

File: src/OpenLoco/src/Map/AnimationManager.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Map/AnimationManager.cpp:66]: It's a very complex update function

File: src/OpenLoco/src/Map/StationElement.h (hunk 3 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Map/StationElement.h:61]: Yes sorry in my head it made sense

**Comment by duncanspumpkin** [comment]
No divergences on both of my replays.

**Pull request merged**